lib/nationsync/auth.rb

```diff
@@ -28,15 +28,26 @@ def authenticate!
 
       # Submit the form
       resp = signin.submit()
+
+      if resp.body.include? "Password and email do not match"
+        puts "Password and email do not match"
+        return false
+      end
       # Returns a redirect we have to follow to finish up the authorization.
-      auth = agent.get resp.header["Location"]
+      puts "Redirect to " + resp.header["location"]
+      auth = agent.get resp.header["location"]
+
+      if auth.header["location"].nil?# app://com.nationbuildertheme/index.html?code=[blah]"
```

Comment:
This comment seems a little misleading, how about one right above the condition explaining what we're doing?

```ruby
# If there's no "Location" header then assume we've been shown an
# authorization confirmation screen and submit the confirmation form
if auth.header['location'].nil?
  ...
```